scripts/envs/open_manipulator/OMReacher_v0.py

```diff
@@ -0,0 +1,368 @@
+#!/usr/bin/env python
+import sys
+import os
+import rospy
+import numpy as np
+import rospkg
+# reads open manipulator's state
+from geometry_msgs.msg import Pose, PoseStamped
+from sensor_msgs.msg import JointState, Image
+from std_msgs.msg import *
+from control_msgs.msg import JointTrajectoryControllerState
+from trajectory_msgs.msg import JointTrajectory, JointTrajectoryPoint
+from gazebo_msgs.msg import ContactsState
+from tf import TransformListener
+import math
+import cv2
+from cv_bridge import CvBridge, CvBridgeError
+from string import Template
+import time
+from random import *
+from geometry_msgs.msg import (
+    PoseStamped,
+    Pose,
+    Point,
+    Quaternion,
+)
+
+from gazebo_msgs.srv import (
+    SetModelState,
+    GetModelState,
+    SpawnModel,
+    DeleteModel,
+)
+
+from gazebo_msgs.srv import (
+    GetModelState
+)
+base_dir = os.path.dirname(os.path.realpath(__file__))
+
+overhead_orientation = Quaternion(
+                         x=-0.00142460053167,
+                         y=0.999994209902,
+                         z=-0.00177030764765,
+                         w=0.00253311793936)
+
+# to read ee position
+from open_manipulator_msgs.msg import *
+
+# episode termination condition
+X_MIN = 0.1
+X_MAX = 0.5
+Y_MIN = -0.3
+Y_MAX = 0.3
+Z_MIN = 0.0
+Z_MAX = 0.6
+TERM_COUNT = 10
+SUC_COUNT = 10
+
+
+ACTION_DIM = 3 # Cartesian
+OBS_DIM = (100,100,3)      # POMDP
+STATE_DIM = 24        # MDP
+
+class robotEnv():
```

Comment:
would be better if you inherit the gazebo [env class](https://github.com/erlerobot/gym-gazebo/blob/master/gym_gazebo/envs/gazebo_env.py).
Just override the interface methods.

+ add the necessary requirement as well.

Reply:
If you wanna implement from the initial state, use this one.
[gym.Env](https://github.com/openai/gym/blob/master/gym/core.py).

Usage: https://github.com/openai/gym/blob/master/gym/envs/atari/atari_env.py